### PR TITLE
Misc JIT improvements

### LINKED ISSUE #374: `dual_quicksort_smoke` test failure
Happens with attached proptest file in `proptest-regressions/trace/consolidation/tests/proptests.txt`:

[proptests.txt](https://github.com/vmware/database-stream-processor/files/11238746/proptests.txt)

### PR DESCRIPTION
**Pull request by Kixiron**
* Fixed #374 with more robust checks
* Implemented the following functions:
  - `@dbsp.str.is_nfc()`
  - `@dbsp.str.is_nfd()`
  - `@dbsp.str.is_nfkc()`
  - `@dbsp.str.is_nfkd()`
  - `@dbsp.str.is_lowercase()`
  - `@dbsp.str.is_uppercase()`
  - `@dbsp.str.bit_length()`
  - `@dbsp.str.byte_length()`
  - `@dbsp.str.char_length()`
* Changed the JSON repr of `Neg` cc @mbudiu-vmw 
  - Now has the `layout` field instead of `output_layout` which contains a `StreamLayout`

### CODE REVIEW

File: crates/dbsp/src/trace/consolidation/tests/mod.rs [hunk 1]
```diff
@@ -158,42 +162,137 @@ fn consolidate_paired_slices_corpus() {
...
+    {
+        if key > next_key {
+            panic!("tuple at index {idx} isn't sorted: ({key:?}, {value:?}) > ({next_key:?}, {next_value:?})");
+        }
+    }
+
+    buffer.sort();
+    expected.sort();
+
+    for ((key, value), (expected_key, expected_value)) in buffer.iter().zip(expected) {
```
Comment (ryzhyk) [inline crates/dbsp/src/trace/consolidation/tests/mod.rs:195]: Isn't this just `assert_eq(buffer, expected)`?

Comment (Kixiron) [reply crates/dbsp/src/trace/consolidation/tests/mod.rs:195]: Yes, but it gives a little more detail by telling what the actual incorrect key or value is

**Review by ryzhyk — APPROVED** [review]

**Comment by Kixiron** [comment]
@lalithsuresh Any idea what's going on with the `build-and-push-image` job? It's giving a `curl: (6) Could not resolve host: dbsp` error

**Comment by ryzhyk** [comment]
> @lalithsuresh Any idea what's going on with the build-and-push-image job? It's giving a curl: (6) Could not resolve host: dbsp error

I believe he's already looking into it, and it's unrelated to this PR.

**Comment by lalithsuresh** [comment]
You can ignore it for now. It's from the migration to pg-embed in #375. I'm fixing it in a separate PR.

**Comment by Kixiron** [comment]
@ryzhyk the issue is that I can't merge since the job is failing

**Pull request merged**